pydoctor/epydoc2stan.py

```diff
@@ -474,8 +475,10 @@ def resolve_types(self) -> None:
     def format(self) -> Tag:
         r: List[Tag] = []
 
-        r += format_desc_list('Parameters', self.parameter_descs)
-        if self.return_desc:
+        # Only include parameter or return sections if any are documented
+        if any(p.body for p in self.parameter_descs):
```

Comment:
There is something that bothers me, here. (see other comment). 

We need a way to determine of the type info is coming from AST, in which case we should display it only if it's documented or if the type is coming from docstring fields, in which case it should display anyhow (with undocumented notes). In terms of implementation, the `resolve_types()` might need an adjustment in order to track if the `FieldDesc` is create from `resolve_types()` (meaning we only have AST info), for instance by creating `_BaseFieldDesc` superclass of `FieldDesc` and `AstFieldDesc` and use that instead from `resolve_types()`. Then check for the existence of `FieldDesc` instances in `self.parameter_descs`. 

Tell me if you think it could work,

Talk to you later,

Reply:
This might now work since we want to track the origin of the type info, and this would only work to track undocumented params…

Reply:
> We need a way to determine of the type info is coming from AST, in which case we should display it only if it's documented or if the type is coming from docstring fields, in which case it should display anyhow (with undocumented notes).

This seems a bit off. If someone has chosen to use `@type` (maybe to help their IDE or whatever) but have chosen not to use `@param`, they are choosing to not document, right?

It seems you are suggested that `@type` be treated as an empty `@param` doc? How would I, as a user that doesn't use `@type`, be able to get that empty documentation functionality the way those that _do_ use `@type` get?

Reply:
> because we include them right now, and would ignore them in the next version, this would be a breaking change

This to me is enough to justify it. So to confirm - Include docs if `@type` _or_ `@param` present, otherwise no docs if neither present (regardless if type hinting). Sound right? I can definitely add the type origin on the desc there.